dissect/cstruct/expression.py

```diff
@@ -1,84 +1,284 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Dict
+from typing import TYPE_CHECKING, Callable, Dict
+
+from dissect.cstruct.exceptions import ExpressionParserError
 
 if TYPE_CHECKING:
     from dissect.cstruct import cstruct
 
 
+class ExpressionTokenizer:
+    def __init__(self, expression: str):
+        self.expression = expression
+        self.i: int = 0
```

Comment:
```suggestion
        self.i = 0
```
I think this type hint is redundant, also perhaps consider a more descriptive name for `i`, i.e. `pos` or `idx`.